pkg/pquerier/collector.go

```diff
@@ -263,3 +273,69 @@ func aggregateClientAggregatesCrossSeries(ctx *selectQueryContext, res *qryResul
 	lastT, lastV := it.At()
 	return lastT, lastV, nil
 }
+
+func intervalAggregation(ctx *selectQueryContext, res *qryResults, t int64, v float64) {
+	currentCell := (t - ctx.queryParams.From) / res.query.aggregationParams.Interval
+
+	for _, col := range res.frame.columns {
+		if col.GetColumnSpec().metric == res.name {
+			_ = res.frame.setDataAt(col.Name(), int(currentCell), v)
+		}
+	}
+}
+
+func windowAggregation(ctx *selectQueryContext, res *qryResults, t int64, v float64) {
+	currentCell := (t - ctx.queryParams.From) / res.query.aggregationParams.Interval
+	if (t-ctx.queryParams.From)%res.query.aggregationParams.Interval > 0 {
+		currentCell++
+	}
+
+	if res.query.aggregationParams.GetAggregationWindow() > res.query.aggregationParams.Interval {
+		numEffectedCells := (t + res.query.aggregationParams.GetAggregationWindow() - ctx.queryParams.From - currentCell*res.query.aggregationParams.Interval) / res.query.aggregationParams.Interval
```

Comment:
Simplify? Also, could help to alias stuff like `res.query.aggregationParams.GetAggregationWindow()` to form a more readable expression.